CONTRIBUTING.md

```diff
@@ -0,0 +1,101 @@
+# Contributing to Miden
+
+#### First off, thanks for taking the time to contribute!
+
+You can find a detailed explanation of main project concepts in [docs](https://maticnetwork.github.io/miden/).
+
+We want to make contributing to this project as easy and transparent as possible, whether it's:
+
+- Reporting a [bug](https://github.com/maticnetwork/miden/issues/new)
+- Take part in [discussions](https://github.com/maticnetwork/miden/discussions)
+- Submitting a fix
+- Proposing new [features](https://github.com/maticnetwork/miden/issues/new)
+
+&nbsp;
+
+## Flow
+We are using [Github Flow](https://guides.github.com/introduction/flow/index.html), so all code changes happen through pull requests.
```

Comment:
For w/e reason this link takes me to `https://docs.github.com/en`. Should we use a different link? e.g.: https://docs.github.com/get-started/quickstart/github-flow

Reply:
I agree with this. I also wonder if we should include a link to the quickstart on forking (https://docs.github.com/en/get-started/quickstart/fork-a-repo). It includes info about setting the upstream & staying in sync, which provides more detail related to the rebasing discussed later. We could amend the text to say "... so all code changes happen through pull requests from a [forked repo](https://docs.github.com/en/get-started/quickstart/fork-a-repo)."

I know this is fairly obvious, but forking isn't mentioned in the other github flow link, and you never know when we'll be someone's first open source project :)

Whatever you guys think, though

Reply:
I agree with this. I also wonder if we should include a link to the quickstart on forking (https://docs.github.com/en/get-started/quickstart/fork-a-repo). It includes info about setting the upstream & staying in sync, which provides more detail related to the rebasing discussed later. We could amend the text to say "... so all code changes happen through pull requests from a [forked repo](https://docs.github.com/en/get-started/quickstart/fork-a-repo)."

I know this is fairly obvious, but forking isn't mentioned in the other github flow link, and you never know when we'll be someone's first open source project :)

Whatever you guys think, though

Reply:
Github flow link updated and forking guide link added. I think that links with additional information always a good idea in guides. Especially when we are not losing anything as a tradeoff source.

Reply:
Github flow link updated and forking guide link added. I think that links with additional information always a good idea in guides. Especially when we are not losing anything as a tradeoff source.